applications/GeoMechanicsApplication/custom_processes/set_parameter_field_process.cpp

```diff
@@ -0,0 +1,143 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Aron Noordam
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "set_parameter_field_process.hpp"
+
+#include <utilities/function_parser_utility.h>
+#include <utilities/mortar_utilities.h>
+
+#include "utilities/interval_utility.h"
+#include "geo_mechanics_application_variables.h"
+
+
+namespace Kratos
+{
+
+    SetParameterFieldProcess::SetParameterFieldProcess(ModelPart& rModelPart,
+                                                            Parameters Settings)
+                                                            : mrModelPart(rModelPart),
+                                                            mParameters(Settings)
+{
+    // function type: python, cpp, input
+    Parameters default_parameters(R"(
+        {
+            "help"            : "This process applies a moving load condition belonging to a modelpart. The load moves over line elements.",
+            "model_part_name" : "please_specify_model_part_name",
+            "variable_name"   : "CUSTOM",
+            "func_type"       : "input",               
+            "function"        : "0",
+            "dataset"         : "dummy"
+        }  )"
+    );
+    Parameters mParameters;
+
+    mParameters.RecursivelyValidateAndAssignDefaults(default_parameters);
+
+}
+
+
+void SetParameterFieldProcess::SetValueAtElement(Element& rElement, const Variable<double>& rVar, double Value)
+{
+
+
+    Properties& r_prop = rElement.GetProperties();
```

Comment:
Since you don't intend to use this reference for modifying the referenced object, I would prefer to make this a reference-to-const. In addition, I would also use `auto` in this case since it is better to program against interfaces then against concrete types:
```suggestion
    const auto& r_prop = rElement.GetProperties();
```

In fact, I would suggest to follow the so-called AAA Style (Almost Always Auto). Herb Sutter explains this style [here](https://herbsutter.com/2013/08/12/gotw-94-solution-aaa-style-almost-always-auto/) and he also provides some strong arguments in favor of it. [Herb Sutter is one of the members of the C++ Technical Committee.]